File: cpp/Petshop.cpp
#include <bits/stdc++.h>

using namespace std;

// kelas petshop
class Petshop
{
private:
    // atribut-atribut private
    int id;
    string name;
    string category;
    int price;

public:
    // konstruktor kosong
    Petshop()
    {
        this->id = -1;
        this->name = "";
        this->category = "";
        this->price = -1;
    }

    // konstruktor dengan parameter untuk mengisi atribut
    Petshop(int id, string name, string category, int price)
    {
        this->id = id;
        this->name = name;
        this->category = category;
        this->price = price;
    }

    // getter dan setter (ambil dan set atribut)
    // setter id
    void setId(int id)
    {
        this->id = id;
    }

    // getter id
    int getId()
    {
        return id;
    }

    // setter name
    void setName(string name)
    {
        this->name = name;
    }

    // getter name
    string getName()
    {
        return name;
    }

    // setter category
    void setCategory(string category)
    {
        this->category = category;
    }

    // getter category
    string getCategory()
    {
        return category;
    }

    // setter price
    void setPrice(int price)
    {
        this->price = price;
    }

    // getter price
    int getPrice()
    {
        return price;
    }

    ~Petshop()
    {
    }
};

File: java/Petshop.java
class Petshop {

    // atribut-atribut private
    private int id;
    private String name;
    private String category;
    private int price;

    // constructor kosong
    Petshop() {
        this.id = -1;
        this.name = "";
        this.category = "";
        this.price = -1;
    }

    // constructor dengan parameter
    Petshop(int id, String name, String category, int price) {
        this.id = id;
        this.name = name;
        this.category = category;
        this.price = price;
    }

    // getter dan setter untuk id
    public void setId(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    // getter dan setter untuk name
    public void setName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    // getter dan setter untuk name
    public void setCategory(String category) {
        this.category = category;
    }

    public String getCategory() {
        return category;
    }

    // getter dan setter untuk price
    public void setPrice(int price) {
        this.price = price;
    }

    public int getPrice() {
        return price;
    }
}
